Add Brainfuck parser example

DevYatsu opened the pull request:
A parser is not a parser if it does not parse brainfuck, right ? So i implemented a concrete and simple implementation of a brainfuck parser.

This PR adds a fully working and minimal Brainfuck parser example.
Everything, including the grammar, lexer, parser, and a basic test, is located in the examples/brainfuck/ folder.

Feel free to review and suggest improvements!

0x2a-42 commented on examples/brainfuck/src/lexer.rs:
```diff
@@ -0,0 +1,94 @@
+    Plus,
+    #[token("-")]
+    Minus,
+    #[token("<")]
+    Left,
+    #[token(">")]
+    Right,
```
I suggest to always use the symbol name for the name of punctuation tokens. In more complicated languages this avoids confusion, when a symbol has different semantic meaning in parts of the grammar.
```suggestion
    #[token("<")]
    Lt,
    #[token(">")]
    Gt,
```

0x2a-42 commented on examples/brainfuck/README.md:
```diff
@@ -0,0 +1,3 @@
+# lelwel-brainfuck
+
+A parser for brainfuck.
```
```suggestion
A parser for [Brainfuck](https://en.wikipedia.org/wiki/Brainfuck).
``

0x2a-42 commented on examples/brainfuck/src/brainfuck.llw:
```diff
@@ -0,0 +1,32 @@
+  RBrack=']'
+  Ignored='<any>';
+
+skip Ignored;
```
```suggestion
skip Comment;
```

0x2a-42 reviewed (changes requested):
Thanks for your contribution. I have a couple of suggestions.

DevYatsu commented:
Okay, I believe everything should be in order now.

The test wasn’t working properly before, I added wasm_bindgen as done in the other examples to fix it. I also added the Brainfuck example to the workspace members, so it builds correctly as part of the whole project.

I guess the Brainfuck language could also be added to the Lelwel playground in the future, thanks to wasm_bindgen.

0x2a-42 commented on examples/brainfuck/generated.rs:
This generated file was probably committed by accident.

DevYatsu replied:
Yes indeed haha

DevYatsu replied:
Corrected btw
